evm/src/contracts/CircleIntegration/Logic.sol

```diff
@@ -0,0 +1,227 @@
+            transferParams.mintRecipient,
+            wormholeNonce,
+            payload
+        );
+    }
+
+    /// @inheritdoc ICircleIntegration
+    function redeemTokensWithPayload(RedeemParameters calldata params)
+        public
+        returns (DepositWithPayload memory deposit)
+    {
+        require(evmChain() == block.chainid, "invalid evm chain");
+
+        IWormhole.VM memory vaa;
+        (
+            vaa,
+            deposit.token,
+            deposit.amount,
+            deposit.sourceDomain,
+            deposit.targetDomain,
+            deposit.nonce,
+            deposit.fromAddress,
+            deposit.mintRecipient,
+            deposit.payload
+        ) = verifyVaaAndMintLegacy(params.cctpMessage, params.cctpAttestation, params.encodedVaa);
+
+        // NOTE: Reverting with Error(string) comes from the old implementation, so we preserve it.
+        require(
+            vaa.emitterAddress != 0
+                && vaa.emitterAddress == getRegisteredEmitters()[vaa.emitterChainId],
+            "unknown emitter"
+        );
+
+        mapping(bytes32 => bool) storage consumedVaas = getConsumedVaas();
+
+        // Revert if this message has been consumed already. This check is meant to prevent replay
+        // attacks, but it may not be necessary because the CCTP Message Transmitter already keeps
+        // track of used nonces.
+        // NOTE: Reverting with Error(string) comes from the old implementation, so we preserve it.
+        require(!consumedVaas[vaa.hash], "message already consumed");
+
+        // Mark as consumed.
+        consumedVaas[vaa.hash] = true;
+
+        // Emit Redeemed event.
+        emit Redeemed(vaa.emitterChainId, vaa.emitterAddress, vaa.sequence);
+    }
+
+    // getters
+
+    /// @inheritdoc ICircleIntegration
+    function fetchLocalTokenAddress(uint32 remoteDomain, bytes32 remoteToken)
+        public
+        view
+        returns (bytes32)
+    {
+        return fetchLocalToken(remoteDomain, remoteToken);
+    }
+
+    /// @inheritdoc ICircleIntegration
+    function addressToBytes32(address evmAddr) public pure returns (bytes32 converted) {
+        converted = evmAddr.toUniversalAddress();
+    }
+
+    /// @inheritdoc ICircleIntegration
+    function decodeDepositWithPayload(bytes memory encoded)
+        public
+        pure
+        returns (DepositWithPayload memory deposit)
+    {
+        // This is a hack to get around using the decodeDeposit method. This is not a real VM
+        // obviously.
+        //
+        // Plus, this getter should never be used in practice.
+        IWormhole.VM memory fakeVaa;
+        fakeVaa.payload = encoded;
+        (
+            deposit.token,
+            deposit.amount,
+            deposit.sourceDomain,
+            deposit.targetDomain,
+            deposit.nonce,
+            deposit.fromAddress,
+            deposit.mintRecipient,
+            deposit.payload
+        ) = fakeVaa.decodeDeposit();
+    }
```

Comment:
Do we need this still? Same with `encodeDepositWithPayload` below?

Reply:
I was afraid to remove these because the contacts deployed already have these methods exposed (so I am not sure if they are used by a contract composing with `CircleIntegration`). It seems unlikely though, so I'm fine to remove these methods.

@gator-boi how do you feel about this?

Reply:
Unfortunately, we need to keep these around. I think it's generally good practice to keep any non-gated public methods in the implementation.

Reply:
Yeah no strong feelings on this so happy to keep around for continuity

Reply:
I also recall that back in the day we put this in so someone could use Etherscan to call these getters.

We'll keep these getters and mark this as resolved @djb15